### PR TITLE
fix(editor): Follow up fixes and improvements to viewer role

### PR DESCRIPTION
**Pull request by r00gm**
## Summary

Added readonly variant for viewer role

## Related Linear tickets, Github issues, and Community forum posts

https://linear.app/n8n/issue/N8N-7626/plus-node-button-should-not-be-visible-on-demo-view-and-executions

## Review / Merge checklist

- [ ] PR title and summary are descriptive. ([conventions](../blob/master/.github/pull_request_title_conventions.md)) 
- [ ] [Docs updated](https://github.com/n8n-io/n8n-docs) or follow-up ticket created.
- [x] Tests included. 
- [ ] PR Labeled with `release/backport` (if the PR is an urgent fix that needs to be backported)

### CODE REVIEW

**Review by cstuncsik — CHANGES_REQUESTED** [review]

File: packages/editor-ui/src/components/executions/workflow/WorkflowExecutionsPreview.test.ts [hunk 1]
```diff
@@ -119,4 +119,34 @@ describe('WorkflowExecutionsPreview.vue', () => {
...
 	);
+
+	it('disables teh stop execution button when the user cannot update', () => {
+		settingsStore.settings.enterprise = {
+			...(settingsStore.settings.enterprise ?? {}),
+		};
+		vi.spyOn(workflowsStore, 'getWorkflowById').mockReturnValue({
+			scopes: undefined,
+		} as IWorkflowDb);
+		const { getByTestId } = render(WorkflowExecutionsPreview, {
```
[the remark this answers is not in the document]
Comment (r00gm) [reply packages/editor-ui/src/components/executions/workflow/WorkflowExecutionsPreview.test.ts:130]: i'll try tomorrow first thing

[the remark this answers is not in the document]
Comment (r00gm) [reply packages/editor-ui/src/components/executions/workflow/WorkflowExecutionsPreview.test.ts]: done!

**Review by cstuncsik — APPROVED** [review]

File: packages/editor-ui/src/components/executions/workflow/WorkflowExecutionsPreview.test.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (cstuncsik) [reply packages/editor-ui/src/components/executions/workflow/WorkflowExecutionsPreview.test.ts]: Thanks

**Comment by Joffcom** [comment]
Got released with [`n8n@1.59.0`](https://github.com/n8n-io/n8n/releases/tag/n8n%401.59.0)

**Pull request merged**